### Sommaire

* [Chapitre 1. Premières explorations des sets de données](#chapter1)
    * [Section 1.1. Contexte et objectifs du projet](#section_1_1)
    * [Section 1.2. Importation des librairies](#section_1_2)
    * [Section 1.3. Importation des données](#section_1_3)
    * [Section 1.4. Exploration préliminaire des variables du jeu de données](#section_1_4)
        * [Section 1.4.1. Analyse des types et des colonnes, et forme des DataFrames ](#section_1_4_1)
        * [Section 1.4.2. Analyse des doublons dans chaque DataFrame](#section_1_4_2)
        * [Section 1.4.3. Résumé de la qualité des données](#section_1_4_3)
* [Chapitre 2. Exploration et Analyse de données avec DataViz](#chapter2)
    * [Section 2.1](#section_2_1)
    * [Section 2.2](#section_2_2)
    * [Section 2.3](#section_2_3)
    * [Section 2.4](#section_2_4)
    * [Section 2.5](#section_2_5)
 
* [Chapitre 3. Nettoyage de données et Pre-procession](#chapter3)
    * [Section 3.1](#section_3_1)
    * [Section 3.2](#section_3_2)
    * 
* [Chapitre 4. ](#chapter4)
    * [Section 4.1](#section_4_1)
    * [Section 4.2](#section_4_2)


### Chapitre 1. Premières explorations des sets de données <a class="anchor" id="chapter1"></a>

#### Section 1.1. Contexte et objectifs du projet <a class="anchor" id="section_1_1"></a>

**Contexte :** L'INSEE, Institut national de la statistique et des études économiques, est l'organisme officiel français chargé de recueillir une variété de données sur le territoire français. Ces données, qu'elles soient démographiques (telles que les naissances, les décès, la densité de la population...) ou économiques (comme les salaires, le nombre d'entreprises par secteur d'activité ou par taille...), offrent une vision complète de la société française. 
Elles constituent ainsi une ressource précieuse pour analyser et comprendre les dynamiques sociales, économiques et territoriales du pays.

**Objectifs :** Cette étude vise à comparer les inégalités en France selon plusieurs dimensions. Tout d'abord, nous nous pencherons sur les disparités entre les entreprises, en examinant leur localisation géographique et leur taille. 
Ensuite, nous nous intéresserons aux inégalités au sein de la population, en analysant les variations de salaires en fonction de différents critères tels que la catégorie d’emploi et la localisation géographique. 
Enfin, nous concentrerons notre attention sur une grande ville en particulier, afin d'étudier de manière approfondie les inégalités qui peuvent exister à l'échelle locale.


#### Section 1.2. Importation des librairies <a class="anchor" id="section_1_2"></a>

In [1]:
import sys
sys.path.append('../src')

import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

# Import modules from the appended path
from fi_functions import *

#### Section 1.3. Importation des données <a class="anchor" id="section_1_3"></a>

In [2]:
df_entreprises = pd.read_csv('../data/base_etablissement_par_tranche_effectif.csv') 
display(df_entreprises.head())

df_salary = pd.read_csv('../data/net_salary_per_town_categories_update2021.csv', sep = ';')
display(df_salary.head())

df_name_geographic = pd.read_csv('../data/name_geographic_information.csv', na_values = '-') 
display(df_name_geographic.head())

df_population = pd.read_csv('../data/population.csv', dtype={'CODGEO': object} )
display(df_population.head())

CODGEO                   LIBGEO  REG DEP  E14TST  E14TS0ND  E14TS1  E14TS6  \
0  01001  L'Abergement-Clémenciat   82  01      25        22       1       2   
1  01002    L'Abergement-de-Varey   82  01      10         9       1       0   
2  01004        Ambérieu-en-Bugey   82  01     996       577     272      63   
3  01005      Ambérieux-en-Dombes   82  01      99        73      20       3   
4  01006                  Ambléon   82  01       4         4       0       0   

   E14TS10  E14TS20  E14TS50  E14TS100  E14TS200  E14TS500  
0        0        0        0         0         0         0  
1        0        0        0         0         0         0  
2       46       24        9         3         2         0  
3        1        2        0         0         0         0  
4        0        0        0         0         0         0

CODGEO     SNHM20    SNHMC20    SNHMP20    SNHME20    SNHMO20    SNHMF20  \
0   1004  15.013132  25.221939  16.950395  11.259209  11.772666  13.043695   
1   1007  15.261333  24.391671  16.179052  11.917058  12.520221  13.855898   
2   1014  14.578709  27.940065  16.384056  12.175141  11.848787  12.570863   
3   1024  14.658174  23.817275  16.319296  11.905547  12.127913  13.053780   
4   1025  14.952910  25.822076  15.365464  11.663794  12.400297  13.444614   

    SNHMFC20   SNHMFP20   SNHMFE20  ...   SNHMHO20   SNHM1820   SNHM2620  \
0  21.806580  14.368130  11.045518  ...  12.095672  10.880315  15.072711   
1  21.532587  15.160246  11.951341  ...  12.881610  10.544073  15.263487   
2  20.702968  13.198429  12.152977  ...  12.304387  11.058325  14.116861   
3  20.157674  14.923592  11.414435  ...  12.525517  10.498073  14.627982   
4  22.271103  14.233353  11.410936  ...  12.666390  10.385396  14.750372   

    SNHM5020  SNHMF1820  SNHMF2620  SNHMF5020  SNHMH1820  SNHMH2620  SNHMH5020  
0  17.255747  10.291297  13.234401  14.192028  11.286232  16.130620  19.226540  
1  16.961952  10.618428  14.034526  14.418987  10.487221  16.229187  19.324302  
2  16.528222  10.295307  12.226430  13.932078  11.623660  15.517659  18.409024  
3  16.404495  10.031113  13.268833  13.691905  10.812016  15.505663  18.422798  
4  16.854860  10.309374  13.513608  14.336143  10.442608  15.580433  18.421533  

[5 rows x 25 columns]

EU_circo  code_région   nom_région chef.lieu_région numéro_département  \
0  Sud-Est           82  Rhône-Alpes             Lyon                 01   
1  Sud-Est           82  Rhône-Alpes             Lyon                 01   
2  Sud-Est           82  Rhône-Alpes             Lyon                 01   
3  Sud-Est           82  Rhône-Alpes             Lyon                 01   
4  Sud-Est           82  Rhône-Alpes             Lyon                 01   

  nom_département       préfecture  numéro_circonscription  \
0             Ain  Bourg-en-Bresse                       1   
1             Ain  Bourg-en-Bresse                       1   
2             Ain  Bourg-en-Bresse                       1   
3             Ain  Bourg-en-Bresse                       1   
4             Ain  Bourg-en-Bresse                       1   

            nom_commune codes_postaux  code_insee   latitude longitude  \
0              Attignat         01340        1024  46.283333  5.166667   
1              Beaupont         01270        1029  46.400000  5.266667   
2                  Bény         01370        1038  46.333333  5.283333   
3             Béreyziat         01340        1040  46.366667      5.05   
4  Bohas-Meyriat-Rignat         01250        1245  46.133333       5.4   

   éloignement  
0         1.21  
1         1.91  
2         1.51  
3         1.71  
4         1.01

NIVGEO CODGEO                   LIBGEO  MOCO  AGEQ80_17  SEXE  NB
0    COM  01001  L'Abergement-Clémenciat    11          0     1  15
1    COM  01001  L'Abergement-Clémenciat    11          0     2  15
2    COM  01001  L'Abergement-Clémenciat    11          5     1  20
3    COM  01001  L'Abergement-Clémenciat    11          5     2  20
4    COM  01001  L'Abergement-Clémenciat    11         10     1  20

#### Section 1.4. Exploration préliminaire des variables du jeu de données <a class="anchor" id="section_1_4"></a> 

##### Section 1.4.1. Analyse des types et des colonnes, et forme des DataFrames <a class="anchor" id="section_1_4_1"></a> 

In [3]:
# Review data types and columns info and shape

print(df_entreprises.info())
print(df_salary.info())
print(df_name_geographic.info())
print(df_population.info())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 36681 entries, 0 to 36680
Data columns (total 14 columns):
 #   Column    Non-Null Count  Dtype 
---  ------    --------------  ----- 
 0   CODGEO    36681 non-null  object
 1   LIBGEO    36681 non-null  object
 2   REG       36681 non-null  int64 
 3   DEP       36681 non-null  object
 4   E14TST    36681 non-null  int64 
 5   E14TS0ND  36681 non-null  int64 
 6   E14TS1    36681 non-null  int64 
 7   E14TS6    36681 non-null  int64 
 8   E14TS10   36681 non-null  int64 
 9   E14TS20   36681 non-null  int64 
 10  E14TS50   36681 non-null  int64 
 11  E14TS100  36681 non-null  int64 
 12  E14TS200  36681 non-null  int64 
 13  E14TS500  36681 non-null  int64 
dtypes: int64(11), object(3)
memory usage: 3.9+ MB
None
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5421 entries, 0 to 5420
Data columns (total 25 columns):
 #   Column     Non-Null Count  Dtype  
---  ------     --------------  -----  
 0   CODGEO     5421 non-null   object 


In [4]:
# Shape for each DataFrame 

print("Shape of DF:")
print("entreprise:",df_entreprises.shape)
print("salary:",df_salary.shape)
print("name_geographic:",df_name_geographic.shape)
print("population:",df_population.shape)

Shape of DF:
entreprise: (36681, 14)
salary: (5421, 25)
name_geographic: (36840, 14)
population: (8536584, 7)


##### Section 1.4.2. Analyse des doublons dans chaque DataFrame<a class="anchor" id="section_1_4_2"></a> 

In [5]:
# Number of duplicates for each DataFrame

print('Number of duplicates :')
print('entreprises :', df_entreprises.duplicated().sum())
print('salary :', df_salary.duplicated().sum())
print('name_geographic :', df_name_geographic.duplicated().sum(), ', name_geographic by code_insee :', df_name_geographic.duplicated(subset=['code_insee']).sum())
print('population :', df_population.duplicated().sum())

Number of duplicates :
entreprises : 0
salary : 0
name_geographic : 0 , name_geographic by code_insee : 147
population : 0


##### Section 1.4.3. Résumé de la qualité des données<a class="anchor" id="section_1_4_3"></a>  

In [6]:
summary(df_entreprises)
summary(df_salary)
summary(df_population)
summary_short(df_name_geographic)

type_info %_missing_values nb_unique_values nb_zero_values  \
CODGEO      object              0.0            36681              0   
LIBGEO      object              0.0            34142              0   
REG          int64              0.0               27              0   
DEP         object              0.0              101              0   
E14TST       int64              0.0             1423            399   
E14TS0ND     int64              0.0             1125            579   
E14TS1       int64              0.0              650           6118   
E14TS6       int64              0.0              272          20324   
E14TS10      int64              0.0              220          22641   
E14TS20      int64              0.0              160          25884   
E14TS50      int64              0.0               87          30678   
E14TS100     int64              0.0               57          33108   
E14TS200     int64              0.0               42          34460   
E14TS500     int64              0.0               23          35933   

         %_zero_values  list_unique_values    mean_or_mode               flag  
CODGEO             0.0  Too many values...           01001  Nothing to report  
LIBGEO             0.0  Too many values...  Sainte-Colombe  Nothing to report  
REG                0.0  Too many values...       49.415365  Nothing to report  
DEP                0.0  Too many values...              62  Nothing to report  
E14TST             1.0  Too many values...      123.456067  Nothing to report  
E14TS0ND           2.0  Too many values...       83.555301  Nothing to report  
E14TS1            17.0  Too many values...       27.291486  Nothing to report  
E14TS6            55.0  Too many values...         5.22055  Nothing to report  
E14TS10           62.0  Too many values...        3.800333  Nothing to report  
E14TS20           71.0  Too many values...        2.296448  Nothing to report  
E14TS50           84.0  Too many values...        0.738339   It's imbalanced!  
E14TS100          90.0  Too many values...        0.332434   It's imbalanced!  
E14TS200          94.0  Too many values...         0.17276   It's imbalanced!  
E14TS500          98.0  Too many values...        0.048417   It's imbalanced!

type_info %_missing_values nb_unique_values nb_zero_values  \
CODGEO       object              0.0             5421              0   
SNHM20      float64              0.0             5421              0   
SNHMC20     float64              0.0             5421              0   
SNHMP20     float64              0.0             5421              0   
SNHME20     float64              0.0             5421              0   
SNHMO20     float64              0.0             5421              0   
SNHMF20     float64              0.0             5421              0   
SNHMFC20    float64              0.0             5421              0   
SNHMFP20    float64              0.0             5421              0   
SNHMFE20    float64              0.0             5421              0   
SNHMFO20    float64              0.0             5421              0   
SNHMH20     float64              0.0             5421              0   
SNHMHC20    float64              0.0             5421              0   
SNHMHP20    float64              0.0             5421              0   
SNHMHE20    float64              0.0             5421              0   
SNHMHO20    float64              0.0             5421              0   
SNHM1820    float64              0.0             5421              0   
SNHM2620    float64              0.0             5421              0   
SNHM5020    float64              0.0             5421              0   
SNHMF1820   float64              0.0             5421              0   
SNHMF2620   float64              0.0             5421              0   
SNHMF5020   float64              0.0             5421              0   
SNHMH1820   float64              0.0             5421              0   
SNHMH2620   float64              0.0             5421              0   
SNHMH5020   float64              0.0             5421              0   

          %_zero_values  list_unique_values mean_or_mode               flag  
CODGEO              0.0  Too many values...        10003  Nothing to report  
SNHM20              0.0  Too many values...    15.440462  Nothing to report  
SNHMC20             0.0  Too many values...    25.194776  Nothing to report  
SNHMP20             0.0  Too many values...    15.847373  Nothing to report  
SNHME20             0.0  Too many values...    11.573974  Nothing to report  
SNHMO20             0.0  Too many values...    12.124348  Nothing to report  
SNHMF20             0.0  Too many values...    13.824962  Nothing to report  
SNHMFC20            0.0  Too many values...     22.03816  Nothing to report  
SNHMFP20            0.0  Too many values...    14.682978  Nothing to report  
SNHMFE20            0.0  Too many values...    11.402604  Nothing to report  
SNHMFO20            0.0  Too many values...    10.775661  Nothing to report  
SNHMH20             0.0  Too many values...    16.600402  Nothing to report  
SNHMHC20            0.0  Too many values...    26.766175  Nothing to report  
SNHMHP20            0.0  Too many values...    16.815999  Nothing to report  
SNHMHE20            0.0  Too many values...    12.058152  Nothing to report  
SNHMHO20            0.0  Too many values...    12.391048  Nothing to report  
SNHM1820            0.0  Too many values...    10.592003  Nothing to report  
SNHM2620            0.0  Too many values...    15.129761  Nothing to report  
SNHM5020            0.0  Too many values...    17.746603  Nothing to report  
SNHMF1820           0.0  Too many values...    10.246277  Nothing to report  
SNHMF2620           0.0  Too many values...    13.812102  Nothing to report  
SNHMF5020           0.0  Too many values...    15.073498  Nothing to report  
SNHMH1820           0.0  Too many values...     10.83844  Nothing to report  
SNHMH2620           0.0  Too many values...     16.07335  Nothing to report  
SNHMH5020           0.0  Too many values...    19.656564  Nothing to report

type_info %_missing_values nb_unique_values nb_zero_values  \
NIVGEO       object              0.0                1              0   
CODGEO       object              0.0            35868              0   
LIBGEO       object              0.0            33452              0   
MOCO          int64              0.0                7              0   
AGEQ80_17     int64              0.0               17         502152   
SEXE          int64              0.0                2              0   
NB            int64              0.0             2953        6325631   

          %_zero_values            list_unique_values    mean_or_mode  \
NIVGEO              0.0                         [COM]             COM   
CODGEO              0.0            Too many values...           01001   
LIBGEO              0.0            Too many values...  Sainte-Colombe   
MOCO                0.0  [11, 12, 21, 22, 23, 31, 32]              11   
AGEQ80_17           6.0            Too many values...               0   
SEXE                0.0                        [1, 2]               1   
NB                 74.0            Too many values...        7.446743   

                        flag  
NIVGEO      It's imbalanced!  
CODGEO     Nothing to report  
LIBGEO     Nothing to report  
MOCO       Nothing to report  
AGEQ80_17  Nothing to report  
SEXE       Nothing to report  
NB         Nothing to report

nb_unique_values %_missing_values nb_missing_values  \
EU_circo                              8              0.0                 0   
code_région                          28              0.0                 0   
nom_région                           28              0.0                 0   
chef.lieu_région                     28              0.0                 0   
numéro_département                  102              0.0                 0   
nom_département                     102              0.0                 0   
préfecture                          102              0.0                 0   
numéro_circonscription               24              0.0                 0   
nom_commune                       34142              0.0                 0   
codes_postaux                      6106              0.0                 0   
code_insee                        36693              0.0                 0   
latitude                            799              8.0              2929   
longitude                          1149              8.0              2905   
éloignement                         293              8.0              2962   

                           type  
EU_circo                 object  
code_région               int64  
nom_région               object  
chef.lieu_région         object  
numéro_département       object  
nom_département          object  
préfecture               object  
numéro_circonscription    int64  
nom_commune              object  
codes_postaux            object  
code_insee                int64  
latitude                float64  
longitude                object  
éloignement             float64


**Conclusion pour name_geographic DF :** 
- Les noms des colonnes en français, tandis que ceux des autres DataFrames sont en anglais. Il est donc nécessaire de les renommer pour une cohérence.
- La colonne "code_insee" semble correspondre à "CODGEO", mais il y a un '0' au debut, par example, '1024' au lieu de '01024'.
- Les colonnes "latitude", "longitude", "éloignement" présentent respectivement 2929, 2841 et 2962 valeurs manquantes (NaN).
- La colonne "longitude" devrait normalement être de type float64. Cependant, nous n'avons pas pu modifier le type en raison de problèmes avec les données. Par exemple :
  
    - L'utilisation de ',' au lieu de '.' pour la notation décimale : **'5,83'**
    - La présence de '-' : ceux-ci doivent être remplacés par des valeurs NaN.

#### Section 1.5 <a class="anchor" id="section_1_5"></a>

In [7]:
#Maxime ==> Intégrer fichiersvg

### Chapitre 2. Exploration et Analyse de données avec DataViz <a class="anchor" id="chapter2"></a>

#### Section 2.1 <a class="anchor" id="section_2_1"></a>

#### Section 2.2 <a class="anchor" id="section_2_2"></a>

#### Section 2.3 <a class="anchor" id="section_2_3"></a>
#### Section 2.4 <a class="anchor" id="section_2_4"></a>
#### Section 2.5 <a class="anchor" id="section_2_5"></a>





In [8]:
df_name_geographic_final = pd.read_csv('../data/name_geographic_final.csv') 

display(df_name_geographic_final.head())

C:\Users\Caroline\AppData\Local\Temp\ipykernel_12952\3261795419.py:1: DtypeWarning: Columns (0,5,7,14) have mixed types. Specify dtype option on import or set low_memory=False.
  df_name_geographic_final = pd.read_csv('../data/name_geographic_final.csv')


COM_code_insee              COM_nom  COM_latitude  COM_longitude DEPT  \
0          10002           Ailleville     48.254630       4.694387   ||   
1          10002           Ailleville     48.254630       4.694387   ||   
2          10002           Ailleville     48.254630       4.694387   ||   
3          10003  Aix-Villemaur-Pâlis     48.226922       3.745811   ||   
4          10003  Aix-Villemaur-Pâlis     48.226922       3.745811   ||   

  DEPT_code DEPT_nom DEPT_ChefLieu_Code_insee DEPT_ChefLieu  \
0        10     Aube                    10387        troyes   
1        10     Aube                    10387        troyes   
2        10     Aube                    10387        troyes   
3        10     Aube                    10387        troyes   
4        10     Aube                    10387        troyes   

   DEPT_ChefLieu_latitude  ...  REG_ChefLieu_latitude REG_ChefLieu_longitude  \
0               48.296734  ...              48.571265                7.76776   
1               48.296734  ...              48.571265                7.76776   
2               48.296734  ...              48.571265                7.76776   
3               48.296734  ...              48.571265                7.76776   
4               48.296734  ...              48.571265                7.76776   

   CAP Capitale Capitale_latitude Capitale_longitude  DIST  DIST_COM_CL_DEPT  \
0   ||    Paris         48.852937            2.35005    ||             45.82   
1   ||    Paris         48.852937            2.35005    ||             45.82   
2   ||    Paris         48.852937            2.35005    ||             45.82   
3   ||    Paris         48.852937            2.35005    ||             25.82   
4   ||    Paris         48.852937            2.35005    ||             25.82   

  DIST_COM_CL_REG DIST_COM_PARIS  
0          229.52         184.91  
1          229.52         184.91  
2          229.52         184.91  
3          299.33         124.11  
4          299.33         124.11  

[5 rows x 26 columns]

In [20]:
df_name_geographic_final.shape

(249340, 26)

In [19]:
# Verification of duplicates by code_insee in df_name_geographic
df_name_geographic_final[df_name_geographic_final.duplicated(subset = ['COM_code_insee'])]

COM_code_insee              COM_nom  COM_latitude  COM_longitude DEPT  \
1               10002           Ailleville     48.254630       4.694387   ||   
2               10002           Ailleville     48.254630       4.694387   ||   
4               10003  Aix-Villemaur-Pâlis     48.226922       3.745811   ||   
5               10003  Aix-Villemaur-Pâlis     48.226922       3.745811   ||   
6               10003  Aix-Villemaur-Pâlis     48.226922       3.745811   ||   
...               ...                  ...           ...            ...  ...   
249335          97617             Tsingoni    -12.782686      45.134576   ||   
249336          97617             Tsingoni    -12.782686      45.134576   ||   
249337          97617             Tsingoni    -12.782686      45.134576   ||   
249338          97617             Tsingoni    -12.782686      45.134576   ||   
249339          97617             Tsingoni    -12.782686      45.134576   ||   

       DEPT_code DEPT_nom DEPT_ChefLieu_Code_insee DEPT_ChefLieu  \
1             10     Aube                    10387        troyes   
2             10     Aube                    10387        troyes   
4             10     Aube                    10387        troyes   
5             10     Aube                    10387        troyes   
6             10     Aube                    10387        troyes   
...          ...      ...                      ...           ...   
249335       976  Mayotte                    97611     mamoudzou   
249336       976  Mayotte                    97611     mamoudzou   
249337       976  Mayotte                    97611     mamoudzou   
249338       976  Mayotte                    97611     mamoudzou   
249339       976  Mayotte                    97611     mamoudzou   

        DEPT_ChefLieu_latitude  ...  REG_ChefLieu_latitude  \
1                    48.296734  ...              48.571265   
2                    48.296734  ...              48.571265   
4                    48.296734  ...              48.571265   
5                    48.296734  ...              48.571265   
6                    48.296734  ...              48.571265   
...                        ...  ...                    ...   
249335              -12.790232  ...             -12.790232   
249336              -12.790232  ...             -12.790232   
249337              -12.790232  ...             -12.790232   
249338              -12.790232  ...             -12.790232   
249339              -12.790232  ...             -12.790232   

       REG_ChefLieu_longitude  CAP Capitale Capitale_latitude  \
1                    7.767760   ||    Paris         48.852937   
2                    7.767760   ||    Paris         48.852937   
4                    7.767760   ||    Paris         48.852937   
5                    7.767760   ||    Paris         48.852937   
6                    7.767760   ||    Paris         48.852937   
...                       ...  ...      ...               ...   
249335              45.194781   ||    Paris         48.852937   
249336              45.194781   ||    Paris         48.852937   
249337              45.194781   ||    Paris         48.852937   
249338              45.194781   ||    Paris         48.852937   
249339              45.194781   ||    Paris         48.852937   

       Capitale_longitude  DIST  DIST_COM_CL_DEPT DIST_COM_CL_REG  \
1                 2.35005    ||             45.82          229.52   
2                 2.35005    ||             45.82          229.52   
4                 2.35005    ||             25.82          299.33   
5                 2.35005    ||             25.82          299.33   
6                 2.35005    ||             25.82          299.33   
...                   ...   ...               ...             ...   
249335            2.35005    ||              6.58            6.58   
249336            2.35005    ||              6.58            6.58   
249337            2.35005    ||              6.58            6.58   
249338       

In [23]:
df_name_geographic_final=df_name_geographic_final.drop_duplicates(subset=['COM_code_insee'], keep="first")

### Chapitre 3. Nettoyage de données et Pre-procession  <a class="anchor" id="chapter3"></a>

#### Section 3.1. df_name_geographic dataset preprocessing <a class="anchor" id="section_3_1"></a>


In [9]:
# Detection errors in the longitude, latitude columns

print(df_name_geographic.longitude.apply(detection_error).loc[df_name_geographic.longitude.apply(detection_error).notna()].values)
print(df_name_geographic.latitude.apply(detection_error).loc[df_name_geographic.latitude.apply(detection_error).notna()].values)

['5,83' '6,01' '6,24' '5,82' '6,31' '4,77' '0,94' '2,07' '0,97' '1,54'
 '1,67' '1,5' '1,78' '1,69' '1,8' '1,63' '1,8' '1,88' '1,67' '0,89' '1,21'
 '1,23' '0,98' '1,15' '1,31' '1,09' '1,18' '1,5' '1,38' '1,11' '1,71'
 '1,93' '1,17' '1,12' '1,68' '1,64' '1,28' '1,02' '1,17' '1,5' '2,25'
 '2,03' '2,15' '2,43' '3,09' '2,89' '4,76' '0,08' '2,51' '4,33' '-2,24'
 '-3,43' '1,91' '1,3' '-5,1' '1,29' '1,2' '5,28' '5,28' '5,96' '1,74'
 '4,07' '3,89' '3,89' '3,48' '3,47' '3,93' '2,93' '3,93' '3,84' '1,76'
 '3,01' '2,03' '4,9' '4,32' '5,84' '2,55' '-0,75' '1,95' '1,97' '6,74'
 '5,04' '-1,94' '0,25' '0,41' '3,28']
[]


In [10]:
# Replacing commas to dots
df_name_geographic["longitude"] = df_name_geographic["longitude"].apply(lambda x: str(x).replace(',','.'))
df_name_geographic["longitude"] = df_name_geographic["longitude"].astype(float)

#df_name_geographic.dtypes

In [11]:
df_name_geographic[df_name_geographic.duplicated(subset = ['code_insee'])]

EU_circo  code_région                  nom_région  \
2293         Sud-Est           93  Provence-Alpes-Côte d'Azur   
2296         Sud-Est           93  Provence-Alpes-Côte d'Azur   
2361         Sud-Est           93  Provence-Alpes-Côte d'Azur   
2994             Est           21           Champagne-Ardenne   
3909             Est           21           Champagne-Ardenne   
...              ...          ...                         ...   
36503  Île-de-France           11               Île-de-France   
36512  Île-de-France           11               Île-de-France   
36515  Île-de-France           11               Île-de-France   
36679  Île-de-France           11               Île-de-France   
36825      Outre-Mer            6    Saint-Pierre-et-Miquelon   

           chef.lieu_région numéro_département           nom_département  \
2293              Marseille                 06           Alpes-Maritimes   
2296              Marseille                 06           Alpes-Maritimes   
2361              Marseille                 06           Alpes-Maritimes   
2994   Châlons-en-Champagne                 08                  Ardennes   
3909   Châlons-en-Champagne                 10                      Aube   
...                     ...                ...                       ...   
36503                 Paris                 94              Val-de-Marne   
36512                 Paris                 94              Val-de-Marne   
36515                 Paris                 94              Val-de-Marne   
36679                 Paris                 95                Val-d'Oise   
36825          Saint-Pierre                975  Saint-Pierre-et-Miquelon   

                 préfecture  numéro_circonscription            nom_commune  \
2293                   Nice                       2                   Nice   
2296                   Nice                       3                   Nice   
2361                   Nice                       5                   Nice   
2994   Charleville-Mézières                       2   Charleville-Mézières   
3909                 Troyes                       2                 Troyes   
...                     ...                     ...                    ...   
36503               Créteil                       7  Saint-Maur-des-Fossés   
36512               Créteil                      10        Vitry-sur-Seine   
36515               Créteil                      11               Gentilly   
36679                 Cergy                       8              Sarcelles   
36825          Saint-Pierre                       1           SAINT-PIERRE   

      codes_postaux  code_insee   latitude  longitude  éloignement  
2293          06000        6088  43.700000   7.250000         1.08  
2296          06000        6088  43.700000   7.250000         1.08  
2361          06000        6088  43.700000   7.250000         1.08  
2994          08000        8105  49.766667   4.716667         1.00  
3909          10000       10387  48.300000   4.083333         0.34  
...             ...         ...        ...        ...          ...  
36503         94100       94068  48.800000   2.500000         0.64  
36512         94400       94081  48.783333   2.400000         0.90  
36515         94250       94037  48.816667   2.333333         1.92  
36679         95200       95585  49.000000   2.383333         1.51  
36825         97500       97502        NaN        NaN          NaN  

[147 rows x 14 columns]

In [12]:
# Drop duplicates by code_insee in name_geographic
df_name_geographic.drop_duplicates(subset=["code_insee"], keep="first", inplace=True)

In [13]:
# Calculate the length of each string in the 'code_insee' column
df_name_geographic['code_insee_length'] = df_name_geographic['code_insee'].astype(str).str.len()

# Find and print unique lengths to determine the number of digits
unique_lengths = df_name_geographic['code_insee_length'].unique()

print("Unique lengths in 'code_insee' column:", unique_lengths)

# Drop the temporary column
df_name_geographic = df_name_geographic.drop(columns=['code_insee_length'])

Unique lengths in 'code_insee' column: [4 5]


In [14]:
# Convert 'code_insee' column to string type
df_name_geographic['code_insee'] = df_name_geographic['code_insee'].astype(str)

# Add leading zeros to 'code_insee' column to match the format of 'CODGEO', 
# assuming 5 digits in 'code_insee'
df_name_geographic.loc[:, 'code_insee'] = df_name_geographic['code_insee'].str.zfill(5) 


In [15]:
print(df_entreprises.shape, df_name_geographic.shape)

(36681, 14) (36693, 14)


#### Section 3.2. df_entreprises dataset preprocessing <a class="anchor" id="section_3_2"></a>

In [16]:
# Create a new column for the enterprise types

df_entreprises['nb_micro_entreprises'] = df_entreprises['E14TS1'] + df_entreprises['E14TS6']
df_entreprises['nb_small_entreprises'] = df_entreprises['E14TS10'] + df_entreprises['E14TS20']
df_entreprises['nb_medium_entreprises'] = df_entreprises['E14TS50'] + df_entreprises['E14TS100']
df_entreprises['nb_large_entreprises'] = df_entreprises['E14TS200'] + df_entreprises['E14TS500']

In [17]:
summary(df_entreprises)

type_info %_missing_values nb_unique_values  \
CODGEO                   object              0.0            36681   
LIBGEO                   object              0.0            34142   
REG                       int64              0.0               27   
DEP                      object              0.0              101   
E14TST                    int64              0.0             1423   
E14TS0ND                  int64              0.0             1125   
E14TS1                    int64              0.0              650   
E14TS6                    int64              0.0              272   
E14TS10                   int64              0.0              220   
E14TS20                   int64              0.0              160   
E14TS50                   int64              0.0               87   
E14TS100                  int64              0.0               57   
E14TS200                  int64              0.0               42   
E14TS500                  int64              0.0               23   
nb_micro_entreprises      int64              0.0              728   
nb_small_entreprises      int64              0.0              294   
nb_medium_entreprises     int64              0.0              109   
nb_large_entreprises      int64              0.0               51   

                      nb_zero_values %_zero_values  list_unique_values  \
CODGEO                             0           0.0  Too many values...   
LIBGEO                             0           0.0  Too many values...   
REG                                0           0.0  Too many values...   
DEP                                0           0.0  Too many values...   
E14TST                           399           1.0  Too many values...   
E14TS0ND                         579           2.0  Too many values...   
E14TS1                          6118          17.0  Too many values...   
E14TS6                         20324          55.0  Too many values...   
E14TS10                        22641          62.0  Too many values...   
E14TS20                        25884          71.0  Too many values...   
E14TS50                        30678          84.0  Too many values...   
E14TS100                       33108          90.0  Too many values...   
E14TS200                       34460          94.0  Too many values...   
E14TS500                       35933          98.0  Too many values...   
nb_micro_entreprises            5532          15.0  Too many values...   
nb_small_entreprises           20328          55.0  Too many values...   
nb_medium_entreprises          29734          81.0  Too many values...   
nb_large_entreprises           34232          93.0  Too many values...   

                         mean_or_mode               flag  
CODGEO                          01001  Nothing to report  
LIBGEO                 Sainte-Colombe  Nothing to report  
REG                         49.415365  Nothing to report  
DEP                                62  Nothing to report  
E14TST                     123.456067  Nothing to report  
E14TS0ND                    83.555301  Nothing to report  
E14TS1                      27.291486  Nothing to report  
E14TS6                        5.22055  Nothing to report  
E14TS10                      3.800333  Nothing to report  
E14TS20                      2.296448  Nothing to report  
E14TS50                      0.738339   It's imbalanced!  
E14TS100                     0.332434   It's imbalanced!  
E14TS200                      0.17276   It's imbalanced!  
E14TS500                     0.048417   It's imbalanced!  
nb_micro_entreprises        32.512036  Nothing to report  
nb_small_entreprises          6.09678  Nothing to report  
nb_medium_entreprises        1.070772   It's imbalanced!  
nb_large_entreprises         0.221177   It's imbalanced!

#### Section 3.3 <a class="anchor" id="section_3_3"></a>

#### Section 3.4 <a class="anchor" id="section_3_4"></a>

In [18]:
# Merge df_entreprises and df_name_geographic

df_merge_ent_geo = pd.merge(left = df_entreprises, right = df_name_geographic, left_on = 'CODGEO', right_on = 'code_insee')


### Chapitre 4. <a class="anchor" id="chapter4"></a>

#### Section 4.1.  <a class="anchor" id="section_4_1"></a>